### PR TITLE
Remove redundant tokenize code

### PR DESCRIPTION
**Pull request by jpwhite4**
## Description
This change removes the unnecessary duplicate tokenize code. It also
replaces the function with a much simpler one that has full unit test
coverage.

## Motivation and Context
General cleanup in preparation for deep linking support.

## Tests performed
Ran the javascript unit_tests to confirm that the location hashes are parsed correctly. Ran code coverage
tools to confirm the new code has 100% test coverage (although the tools cannot check that all possible states have been hit in the regex state machine).

### CODE REVIEW

File: html/unit_tests/spec/.eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+    "env": {
+        "mocha": true
+    }
```
Comment (plessbd) [inline html/unit_tests/spec/.eslintrc.json]: missing ,

**Comment by plessbd** [comment]
did you do some basic before and after tests to show that the same input got the same output with old and new code?

**Comment by plessbd** [comment]
Will this new tokenize code break any existing code since this one always has the keys in the result?

https://github.com/ubccr/xdmod/blob/feddfebaa82f35b14a637fd4dacf3a46fe7b019c/html/internal_dashboard/js/Dashboard/Viewport.js#L65

https://github.com/ubccr/xdmod/blob/feddfebaa82f35b14a637fd4dacf3a46fe7b019c/html/gui/js/modules/About.js#L130

https://github.com/ubccr/xdmod/blob/4b606aa25ffdd474efe9877a8eba2d63214cb77b/html/index.php#L506

etc

https://github.com/ubccr/xdmod/search?utf8=%E2%9C%93&q=tokenize&type=

**Comment by jpwhite4** [comment]
To answer your questions: 1) Yes I did test , 2) No it won't break anything.

I note that the XDMoD.Dashboard.tokenize function was not modified in any way so this code change does not impact any code that calls it.

**Comment by plessbd** [comment]
Screenshots or it didnt happen? :)

Looks Good To Me

**Pull request merged**